src/ui/server/controllers/insurance/policy-and-export/about-goods-or-services/index.test.ts

```diff
@@ -83,27 +95,106 @@ describe('controllers/insurance/policy-and-export/about-goods-or-services', () =
         }),
         ...pageVariables(refNumber),
         application: mockApplication,
+        countries: mapCountries(mockCountries),
       };
 
       expect(res.render).toHaveBeenCalledWith(TEMPLATE, expectedVariables);
     });
 
+    describe('when a final destination has been previously submitted', () => {
+      const mockApplicationWithCountry = {
+        ...mockApplication,
+        policyAndExport: {
+          ...mockApplication.policyAndExport,
+          [FINAL_DESTINATION]: countryIsoCode,
+        },
+      };
+
+      beforeEach(() => {
+        res.locals.application = mockApplicationWithCountry;
+      });
+
+      it('should render template with countries mapped to submitted country', async () => {
+        await get(req, res);
+
+        const expectedVariables = {
+          ...insuranceCorePageVariables({
+            PAGE_CONTENT_STRINGS: PAGES.INSURANCE.POLICY_AND_EXPORTS.ABOUT_GOODS_OR_SERVICES,
+            BACK_LINK: req.headers.referer,
+          }),
+          ...pageVariables(refNumber),
+          application: mockApplicationWithCountry,
+          countries: mapCountries(mockCountries, countryIsoCode),
+        };
+
+        expect(res.render).toHaveBeenCalledWith(TEMPLATE, expectedVariables);
+      });
+    });
+
     describe('when there is no application', () => {
       beforeEach(() => {
         res.locals = { csrfToken: '1234' };
       });
 
-      it(`should redirect to ${ROUTES.PROBLEM_WITH_SERVICE}`, () => {
-        get(req, res);
+      it(`should redirect to ${ROUTES.PROBLEM_WITH_SERVICE}`, async () => {
+        await get(req, res);
 
         expect(res.redirect).toHaveBeenCalledWith(ROUTES.PROBLEM_WITH_SERVICE);
       });
     });
+
+    describe('api error handling', () => {
+      describe('when there are no countries returned from the API', () => {
+        beforeEach(() => {
+          // @ts-ignore
```

Comment:
do we need this?

Reply:
Unfortunately yes - have this in a few places. We are testing here that if the API call is successful, but returns an empty body, we redirect to "problem with service" page.

Typescript does not allow an empty `Promise.resolve()` but here we are stubbing this to mock an empty body from the API

Happy to remove these tests, but feels worth having? We had issues like this with the CIS API before so introduced this test, have been using for all API calls since.